machine-learning-scikit-learn.ipynb

```diff
@@ -0,0 +1,333 @@
+    "\n",
+    "It is important to specify the scoring parameter in `Incremental`; otherwise, scikit-learn scorers are fed Dask arrays (which they're not optimized for).\n",
+    "\n",
+    "\n",
+    "[1]:http://scikit-learn.org/stable/modules/scaling_strategies.html\n",
+    "[inc]:http://scikit-learn.org/stable/modules/scaling_strategies.html#incremental-learning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model training\n",
+    "For an `Incremental` model, `partial_fit` and `fit` aliased to the same item and do one complete pass over the dataset. Let's call `partial_fit` a couple times, and score it after it's called each time (which will allow for a visualization):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "# takes about 1min\n",
+    "\n",
+    "from distributed.metrics import time\n",
+    "data = []\n",
+    "start = time()\n",
+    "for iteration in range(10):\n",
+    "    inc.partial_fit(X_train, y_train, classes=da.unique(y))\n",
+    "    data += [{'score': inc.score(X_test, y_test),\n",
+    "             'epochs': iteration + 1,\n",
+    "             'time': time() - start}]\n",
+    "    print(data[-1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "df = pd.DataFrame(data)\n",
+    "df.plot(x='epochs', y='score')\n",
+    "plt.title('Score over iterations')\n",
+    "plt.ylabel('Score')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Predicting\n",
+    "Let's predict, than compute the accuracy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_hat = inc.predict(X_test)\n",
+    "errors = np.abs(y_hat - y_test) / 2   # divide by 2 to get 0-1 labels\n",
+    "accuracy = 1 - errors.mean()\n",
+    "accuracy.compute()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inc.score(X_test, y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Grid search"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This accuracy might be the best possible, but there a lot of parameters to feed `SGDClassifier` that can drastically change the performance. We'll consider two of the most basic parameters, specifying a different loss function and different amount of penalty. This is the problem of hyperparameter optimization, something Dask-ML supports: https://dask-ml.readthedocs.io/en/latest/hyper-parameter-search.html\n",
```

Comment:
Is the first phrase missing a "not"? Somethying like

"This accuracy might not be the best possible; there are a lot of parameters..."